csharp/extractor/Semmle.Extraction.CSharp/Entities/PreprocessorDirectives/EndRegionDirective.cs

```diff
@@ -0,0 +1,23 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.IO;
+
+namespace Semmle.Extraction.CSharp.Entities
+{
+    internal class EndRegionDirective : PreprocessorDirective<EndRegionDirectiveTriviaSyntax>
+    {
+        public EndRegionDirective(Context cx, EndRegionDirectiveTriviaSyntax trivia)
+            : base(cx, trivia)
+        {
+        }
+
+        protected override void PopulatePreprocessor(TextWriter trapFile)
+        {
+            trapFile.directive_endregions(this);
+        }
+
+        internal static void WriteRegionBlock(Context cx, RegionDirective region, EndRegionDirective endregion)
```

Comment:
Why not add `RegionDirective` to the constructor and make the population part of `PopulatePreprocessor` instead?